tests/check_linux_layout_intel.py

```diff
@@ -0,0 +1,192 @@
+
+def check_sign_key():
+    for i in fw_list:
+        full_path = bundle_base + "/" + i
+        try:
+            fw_bin = sof_ri_info.parse_fw_bin(full_path, False, False)
+        except Exception as e:
+            print("FW parse error for %s: %s" % (full_path, str(e)))
+            pass
+
+        if fw_bin == None:
+            return
+
+        hdr = sof_ri_info.getCssManifest(fw_bin).cdir['css_mft_hdr']
+        for attr in hdr.attribs:
+            if attr.name == "modulus":
+                if "/community" in i:
+                    if attr.val != sof_ri_info.COMMUNITY_KEY and \
+                       attr.val != sof_ri_info.COMMUNITY_KEY2:
+                        raise Exception("FW %s not signed with community key" % full_path)
+
+                elif "/intel-signed" in i:
+                    if attr.val == sof_ri_info.COMMUNITY_KEY or \
+                       attr.val == sof_ri_info.COMMUNITY_KEY2 or \
+                       not attr.val in sof_ri_info.KNOWN_KEYS:
+                        raise Exception("Intel signed firmware signed with unexpected key")
+
+
+def check_ipc_version():
+    for i in fw_list:
+        full_path = bundle_base + "/" + i
+        try:
+            fw_bin = sof_ri_info.parse_fw_bin(full_path, False, False)
+        except Exception as e:
+            print("FW parse error for %s: %s" % (full_path, str(e)))
+            pass
+
+        if fw_bin == None:
+            return
+
+        ipc4 = "sof-ipc4/" in i
+        for comp in fw_bin.components:
+            # This check is not 100% certain. It is theoretically possible
+            # IPC3 firwmare to have AE1 style extended manifest header.
+            if not ipc4 and type(comp).__name__ == "ExtendedManifestAE1":
+                print("WARNING: %s IPC3 firmware but AE1 ext manifest" % full_path)
+
+            if ipc4 and type(comp).__name__ == "ExtendedManifestXMan":
+                raise Exception("WARNING: %s IPC4 firmware but Xman ext manifest" % full_path)
+
+def main():
+    global verbose_mode
+    global bundle_base
+
+    parser = argparse.ArgumentParser(description='SOF firmware binary layout tester.')
+
+    parser.add_argument('-r', '--root', type=str, help="Root of the SOF FW bundle")
+    parser.add_argument('-v', '--verbose', action="store_true")
+
+    args = parser.parse_args()
+    verbose_mode = args.verbose
+    bundle_base = args.root
+    if bundle_base == None:
+        bundle_base = BASE
+
+    if verbose_mode: print("Running tests for SOF FW bundle at %s" % bundle_base)
+
+    # check all FW expected by kernel is found in the FW bundle
+    check_fw_files_kernel_to_bundle()
+
+    # check all files in SOF Intel FW bundle and ensure they are listed in fw_list
+    check_fw_files_bundle_to_kernel()
+
+    # go through all files and ensure prod-signed files are really prod-signed (if doable)
+    if sof_ri_info != None: check_sign_key()
```

Comment:
If you separate better the signature check and the listing check then you don't have this missing import problem anymore. If the user is missing sof_ri_info, then it's on the user not to request the signature check.